src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Expressions.cs

```diff
@@ -1123,6 +1123,8 @@ private void EmitLift(ExpressionType nodeType, Type resultType, MethodCallExpres
                         Label exit = _ilg.DefineLabel();
                         Label exitNull = _ilg.DefineLabel();
                         LocalBuilder anyNull = GetLocal(typeof(bool));
+                        _ilg.Emit(OpCodes.Ldc_I4_0);
+                        _ilg.Emit(OpCodes.Stloc, anyNull);
```

Comment:
You mentioned this fixes a regression from .NET Framework 4.8, but it doesn't have these lines either.  Is this an issue of locals not being automatically cleared by the runtime or something?

Reply:
> You mentioned this fixes a regression from .NET Framework 4.8, but it doesn't have these lines either.

The .NET Core implementation has an optimization to reuse locals (see `GetLocal(typeof(bool))` above). The .NET Framework implementation does not reuse locals. In the repro, the existing local contained a non-zero value.

Reply:
Got it, thanks.